astroquery/esa/hsa/tests/test_hsa_remote.py

```diff
@@ -0,0 +1,179 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import tempfile
+import pytest
+
+import os
+import tarfile
+from requests.exceptions import ChunkedEncodingError
+
+from ..core import HSAClass
+
+spire_chksum = [10233, 10762, 9019, 10869, 3944, 11328, 3921, 10999, 10959, 11342, 10974, 3944, 11335, 11323, 11078, 11321, 11089, 11314, 11108, 6281]
+
+pacs_chksum = [10208, 10755, 8917, 10028, 3924, 3935, 6291]
+
+
+@pytest.mark.remote_data
+class TestHSARemote:
+    tmp_dir = tempfile.TemporaryDirectory()
+    retries = 2
+
+    def access_archive_with_retries(self, f, params):
+        for _ in range(self.retries):
+            try:
+                res = f(**params)
+                return res
+            except ChunkedEncodingError:
+                pass
+        return None
+
+    def test_download_data_observation_pacs(self):
+        obs_id = "1342191813"
+        parameters = {'retrieval_type': "OBSERVATION",
+                      'observation_id': obs_id,
+                      'instrument_name': "PACS",
+                      'product_level': 'LEVEL3',
+                      'cache': False,
+                      'download_dir': self.tmp_dir.name}
+        expected_res = os.path.join(self.tmp_dir.name, obs_id + ".tar")
+        hsa = HSAClass()
+        res = self.access_archive_with_retries(hsa.download_data, parameters)
+        if res is None:
+            pytest.skip("Archive broke the connection {} times, unable to test".format(self.retries))
```

Comment:
Here, and everywhere below: I wonder whether it would be more useful to let this case break the test? Otherwise it would be difficult to pick up if there is a genuine issue (or would you always expect a non None return?)

Reply:
While I was testing these last time I noticed that the connection with the archive was very unstable and, at least for large files, a lot times the connection was broken by the archive. The reason why I decided to skip the tests if the archive is unresponsive is to avoid flakiness in the tests. We always expect a non `None` return from these functions (the `filename`) so it should be ok to check for a `None` value to determine if the connection was broken.

Reply:
Are there any smaller files maybe to work with?

Also, if this stays as is, maybe `xfail` is more appropriate than skip?

Reply:
* I agree, `xfail` suits better here. I've changed it.
* Since the changes to make these tests faster I haven't noticed any connection problems. I would still keep the tests with `xfail` just in case (I don't think it hurts to have it 🙂)